src/compiler/checker.ts

```diff
@@ -6670,6 +6681,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         factory.createKeywordTypeNode(SyntaxKind.NeverKeyword)
                     );
                 }
+                else if (needsModifierPreservingWrapper) {
+                    // and step 3: once the mapped type is reconstructed, create a `ConstraintType extends infer T_1 extends keyof ModifiersType ? {[K in T_1]: Template} : never`
+                    // subtly different from the `keyof` constraint case, by including the `keyof` constraint on the `infer` type parameter, it doesn't rely on the constraint type being itself
+                    // constrained to a `keyof` type to preserve it's modifier-preserving behavior. This is all basically because we preserve modifiers for a wider set of mapped types than

```

Comment:
```suggestion
                    // constrained to a `keyof` type to preserve its modifier-preserving behavior. This is all basically because we preserve modifiers for a wider set of mapped types than
```

double-checking: is the effect of this PR that we now preserve modifiers for these mapped types, and didn't before?

Reply:
without reading the code of the fix itself - yes, the issue that I reported is specifically about the fact that modifiers are lost but they shouldn't be so the fix for that issue should preserve them